cmd/argo/commands/set.go

```diff
@@ -0,0 +1,85 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/argoproj/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/argoproj/argo/cmd/argo/commands/client"
+	workflowpkg "github.com/argoproj/argo/pkg/apiclient/workflow"
+)
+
+type setOps struct {
+	message           string // --message
+	phase             string // --phase
+	outputParameters  string // --output-parameters
+	nodeFieldSelector string // --node-field-selector
+}
+
+func NewSetCommand() *cobra.Command {
+	var (
+		setArgs setOps
+	)
+
+	var command = &cobra.Command{
+		Use:   "set FIELD WORKFLOW SET_TO",
+		Short: "set values to zero or more workflows",
+		Example: `# Set outputs to a node within a workflow:
+
+  argo set outputs my-wf '{"parameter-name": "Hello, world!"}' --node-field-selector displayName=approve
```

Comment:
`argo node set`?

Reply:
I thought about this. My hesitation was that we technically don't expose the term `Node` to the user (it is only an implementation detail). However, I don't have a strong opinion on this and could use `argo node set` with minor convincing.

Reply:
Also, any negative opinions on using JSON as a map for output name to output value? I used this format since it mimics `kubectl patch`

Reply:
good examples